### DISCUSSION TITLE
[5일차-지원] async/await에 대해 설명해보세요.

### DISCUSSION THREAD

**Discussion by smarfy99** [zzangCS/zzangInterview#22] [프론트 면접 질문]
이하 동일

**Comment by ez1n** [comment]
**async & await**

> 비동기식 코드를 동기식으로 표현하여 간단하게 나타내는 것

- 기존의 비동기 처리 방식인 콜백함수의 단점을 보완하기 위해 사용했던 Promise의 단점을 보완하기 위해 도입되었다. (개발자가 읽기 좋은 코드를 작성할 수 있도록 함)
- Promise 객체를 반환한다.
- 동기적인 코드 흐름으로 개발 가능
- 에러처리를 잡기 위해 try/catch를 사용한다.

**Reply by ez1n** [reply]
https://joshua1988.github.io/web-development/javascript/js-async-await/

**Comment by skyhanull** [comment]
async/await 는 비동기 함수를 동기함수터럼 사용하는 것으로써
함수앞에 async를 선언하면 promise 객체가 저절로 생성됩니다

async/await는 비동기 코드가 동기코드처럼 읽히게 해주어 가독성을 좋게 해줍니다
이때 async/await 는 try/catch문을 사용합니다.
